
syntax = "proto3"; 

package traack.v1.tracker;

import "proto/v1/response/status.proto";

option go_package = "github.com/mhannig/traack/service/lib/v1/tracker";

message Customer {
    uint64 id = 1;

    string name = 2;
}

// Customers Service
message GetCustomerRequest {
    uint64 id = 1;
}

message GetCustomerResponse {
    Customer customer = 1;
    traack.v1.response.Status status = 2;
}


message ListCustomersRequest {
}

message ListCustomersResponse {
    repeated Customer customers = 1;
    traack.v1.response.Status status = 2;
}

message CreateCustomerRequest {
    Customer customer = 1;
}

message CreateCustomerResponse {
    Customer customer = 1;
    traack.v1.response.Status status = 2;
}

message DeleteCustomerRequest {
    uint64 id = 10;
}

message DeleteCustomerResponse {
    traack.v1.response.Status status = 10;
}

service CustomerService {
    rpc CreateCustomer(CreateCustomerRequest) 
        returns (CreateCustomerResponse) {}

    rpc GetCustomer(GetCustomerRequest)
        returns (GetCustomerResponse) {}

    rpc ListCustomers(ListCustomersRequest)
        returns (ListCustomersResponse) {}

    rpc DeleteCustomer(DeleteCustomerRequest)
        returns (DeleteCustomerResponse) {}
}


